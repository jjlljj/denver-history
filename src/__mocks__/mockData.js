export const mockBuilding = {
  "id": 603,
  "ldmk_num": 21,
  "ldmk_name": "Navarre Building",
  "aka_name": "Bricker Collegiate Institute",
  "ord_num": 470,
  "ord_year": 1971,
  "address_line1": "1727 Tremont Street",
  "address_line2": "1725 Tremont Place",
  "situs_num": 1725,
  "situs_st": "Tremont",
  "situs_type": "PL",
  "historic_dist": 30,
  "state_hist_num": "5DV.124",
  "year_built": "1880",
  "arch_bldr": null,
  "document": null,
  "photo_link": "https://anschutzcollection.org/wp-content/uploads/2017/10/Navarre_2.png",
  "notes": null,
  "gis_notes": null,
  "description": "This iconic building was designed by Frank E. Edbrooke. It was built as a school for girls, called The Brinker Collegiate Institute after its owners, Joseph and Elizabeth Brinker. They owned and operated the school from 1880 (eventually turning it into a coeducational facility) until 1886. Forced into foreclosure after Joseph Brinker’s death, the building was sold to two of the West’s most notorious gamblers. They turned it into Hotel Richelieu, a gentleman’s club. Only six months later, the new owners lost the building in a card game to two other known gamblers, Ed Chase and Vaso Chucovich. They renamed the building “The Navarre” after King Henry Navarre (1553-1610), a devotee of decadence and high living. The building was turned into a bordello, offering public dining and private gambling on the bottom two floors, and more illicit pleasures on the top two levels. The Brown Palace Hotel was built across the street in 1892, and an underground tunnel was constructed between the two buildings to share coal. Legend says that the tunnel was also used to discreetly transport gentlemen guests between the two buildings. At some point in the past The Navarre’s entrance to the tunnel was sealed off, and the length of the tunnel possibly even filled in with masonry to prevent the street overhead from collapsing. In 1904, Denver Mayor Robert Speer was elected. Bowing to public pressure, he put an end to gambling and prostitution in Denver. The Navarre became a respectable dining club, although legend has it that illicit activity continued well into the 1920s. In the 1990s, The Navarre was renovated and restored it to its Victorian roots and the Anschutz Collection of Western American art moved in.",
  "address_id": null,
  "created_at": "2018-04-11T22:00:39.855Z",
  "updated_at": "2018-04-11T22:00:39.855Z",
  "situs_dir": null,
  "lon": "-104.988045901496",
  "lat": "39.7447201469119"
}

export const mockDistrict = [
  {
    "id": 5,
    "ldmk_num": 103,
    "ldmk_name": "Equitable Building",
    "aka_name": null,
    "ord_num": 453,
    "ord_year": 1977,
    "address_line1": "730 17th Street",
    "address_line2": null,
    "situs_num": 730,
    "situs_st": "17th",
    "situs_type": "ST",
    "historic_dist": 3,
    "state_hist_num": "5DV.121",
    "year_built": "c. 1892-93",
    "arch_bldr": null,
    "document": null,
    "photo_link": "Equitable_Building0.jpg",
    "notes": null,
    "gis_notes": null,
    "description": "Completed in 1892, the Equitable Building is significant for its role in the commercial and political history of Colorado. At the time of its completion, it was attributed with the development of 17th Street as a center of finance and business and has since served as the location for many distinguished businesses and law offices.The nine-story E-shaped bldg. cost $1.5 million prior to the silver crash in 1893. Example of Italian Renaissance Revival style of architecture and was originally owned by Equitable Life Insurance Co.",
    "address_id": null,
    "created_at": "2018-04-16T02:01:00.237Z",
    "updated_at": "2018-04-16T02:01:00.237Z",
    "situs_dir": null,
    "lon": "-104.991938634648",
    "lat": "39.7462977886473"
  },
  {
    "id": 6,
    "ldmk_num": 263,
    "ldmk_name": "Hover/Bromley Building",
    "aka_name": null,
    "ord_num": 271,
    "ord_year": 1996,
    "address_line1": "1348 Lawrence Street",
    "address_line2": "1390 Lawrence Street",
    "situs_num": 1348,
    "situs_st": "Lawrence",
    "situs_type": "ST",
    "historic_dist": 3,
    "state_hist_num": "5DV.1719",
    "year_built": "1901",
    "arch_bldr": null,
    "document": null,
    "photo_link": "W_A_Hover__Company0.jpg",
    "notes": null,
    "gis_notes": null,
    "description": "W. A. Hover & Company building located in Denver, Colorado. The 1901 W.A. Hover & Company Building is associated with the development of the wholesale drug industry form 1901 to 1949. The Hover Co. grew into one of the largest regional drug supply companies serving both Colorado and the surrounding states. The building is also architecturally significant as a good example the early 20th century Commercial Style and for its being one of only two intact examples of a commercial building designed by master architect Robert S. Roeschlaub.",
    "address_id": null,
    "created_at": "2018-04-16T02:01:00.237Z",
    "updated_at": "2018-04-16T02:01:00.237Z",
    "situs_dir": null,
    "lon": "-104.998887978911",
    "lat": "39.7463202027288"
  }
]
