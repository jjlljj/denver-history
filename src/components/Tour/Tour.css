.main-wrap {
  display: flex;
  font-family: 'Open Sans', sans-serif;
  height: 90vh;
  justify-content: space-between;
  min-height: 400px;
  position: relative;
}

@media screen and (max-width: 650px) {
  .main-wrap {
    flex-direction: column-reverse;
  }
}
