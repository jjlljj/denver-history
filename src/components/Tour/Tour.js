import React, { Component } from 'react';
import { connect } from 'react-redux';
import PropTypes from 'prop-types';
import './Tour.css';
import Map from '../Map/Map';
import Header from '../Header/Header';
import Sidebar from '../Sidebar/Sidebar';
import Spinner from '../Spinner/Spinner';
import { addDistrict } from '../../actions';
import { getDistrictBuildings } from '../../helpers/apiHelper.js';
import { Route,  withRouter } from 'react-router-dom';

export class Tour extends Component {

  componentDidMount = () => {
    this.fetchBuildings()
  }

  fetchBuildings = async () => {
    try {
      const district = await getDistrictBuildings(3)
      this.props.addDistrict(district);
    } catch (error) {
      console.log(error);
    };
  }

  renderSpinner = () => {
    const { showLoading } = this.props;

    if ( showLoading ) {
      return <Spinner />;
    };
  }

  renderMap = () => {
    const { district } = this.props;

    if (district.length) {
      return <Map geoJSON={ district } />;
    };
  }

  render() {
    return (
      <div>
        <Header />
        <section className="main-wrap">
          <Sidebar />
          { this.renderSpinner() }
          { this.renderMap() }
        </section>
      </div>
    );
  }
}

const mapStateToProps = ({ showLoading, district }) => ({
  showLoading,
  district
})

const mapDispatchToProps = dispatch => ({
  addDistrict: district => dispatch(addDistrict(district))
})

export default withRouter(connect(mapStateToProps, mapDispatchToProps)(Tour));
