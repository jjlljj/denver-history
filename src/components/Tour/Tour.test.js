/* eslint-disable */
import React from 'react';
import ReactDOM from 'react-dom';
import { shallow } from 'enzyme';
import { Tour } from './Tour';
jest.mock('../../../node_modules/mapbox-gl/dist/mapbox-gl', () => () => ({ Map: () => ({}) }));
jest.mock('../../../node_modules/threebox/', () => () => ({ threebox: jest.fn() }));
jest.mock('../../actions/index', () => () => ({ addDistrict: jest.fn() }));
jest.mock('../../.key.js', () => () => ({ MAPBOX_TOKEN: 'faketoken' }));
import { mockDistrict } from '../../__mocks__/mockdata'

describe('Tour', () => {
  let renderedComponent
  let district
  let addDistrict
  let showLoading

  beforeEach(() => {
    district = []
    addDistrict = jest.fn()
    showLoading = true
    renderedComponent = shallow(<Tour district={district} showLoading={ showLoading } addDistrict={addDistrict}/>, { disableLifecycleMethods: true })

    window.fetch = jest.fn().mockImplementation(() => Promise.resolve({
      status: 200,
      json: () => Promise.resolve(mockDistrict)
    }))
  })

  it('matches snapshot', () => {
    expect(renderedComponent).toMatchSnapshot()
  })

  it('fetchBuildings should call the fetch in getDistrictBuildings', () => {
    expect(window.fetch).not.toHaveBeenCalled()

    renderedComponent.instance().fetchBuildings()

    expect(window.fetch).toHaveBeenCalled()
  })

  it('fetchBuildings should call the action addDistrict', async () => {
    expect(addDistrict).not.toHaveBeenCalled()

    const called = await renderedComponent.instance().fetchBuildings()

    called && expect(addDistrict).toHaveBeenCalled()
  })

})
