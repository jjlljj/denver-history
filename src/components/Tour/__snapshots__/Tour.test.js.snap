// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Tour matches snapshot 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Tour
    addDistrict={[Function]}
    showLoading={true}
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <Header />,
        <section
          className="main-wrap"
>
          <Connect(Sidebar) />
          <Spinner />
</section>,
      ],
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {},
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <Connect(Sidebar) />,
            <Spinner />,
          ],
          "className": "main-wrap",
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {},
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {},
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
        ],
        "type": "section",
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <Header />,
          <section
            className="main-wrap"
>
            <Connect(Sidebar) />
            <Spinner />
</section>,
        ],
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {},
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <Connect(Sidebar) />,
              <Spinner />,
            ],
            "className": "main-wrap",
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {},
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {},
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
          ],
          "type": "section",
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
    "disableLifecycleMethods": true,
  },
}
`;
